packages/appstore/src/stores/wallet-store.ts

```diff
@@ -0,0 +1,24 @@
+import { action, observable } from 'mobx';
+import { WS } from '@deriv/shared';
+import BaseStore from './base-store';
+
+export default class WalletStore extends BaseStore {
+    @observable
+    wallet_names = {};
+
+    @action.bound
+    setWalletNames(data: any) {
+        this.wallet_names = data;
+    }
+
+    @action.bound
+    getWalletNames() {
+        WS.authorized.storage
```

Comment:
we should be calling it everytime the `CreateWallet` component is mounted

Reply:
We should only call this API if we don't have the data in-memory.

Reply:
I guess I did not completely get your comment , I have called  getWalletNames() inside UseEffect of CreateWallet, which is the same right.
We should only call this API if we don't have the data in-memory -- for this purpose only I was using caching the call, any other way in which it is done in other deriv packages so to get an idea of what your are referring. Or did you mean save the value in local storage and then use it from there

Reply:
When we create the first wallet, `CreateWallet` component will be mounted which will call this API. Later on,  `CreateWallet` component will be unmounted when the wizard is completed. 
When we want to create a second wallet account,  `CreateWallet` component  will be mounted again which will call this API again.

`WS.authorized.storage` is not in memory, it stores in local storage and fetches it back.
We already have assigned the data to `this.wallet_names`, when we reassign, there will be unwanted reactions and re-renders which can be avoided.